your-first-app.md

```diff
@@ -0,0 +1,650 @@
+clone/extend it.
+
+```javascript
+app.model({
+  state: {
+    todos: []
+  },
+  reducers: {
+    addTodo: (data, state) => {
+      const todo = Object.assign({}, data, {
+        completed: false
+      })
+      const newTodos = [...state.todos, todo]
+      return { todos: newTodos }
+    }
+  }
+})
+```
+
+Now, every time we add a todo item, it will be stored as
+`{ title: 'Our title', complete: false }`. Let's update the view to show that
+status.
+
+```javascript
+const view = (state, prev, send) => {
+  return html`
+    <div>
+      <form onsubmit=${onSubmit}>
+        <input type="text" placeholder="New item" id="title">
+      </form>
+      <ul>
+        ${state.todos.map((todo) => html`
+          <li>
+            <input type="checkbox" ${todo.completed ? 'checked' : ''} />
+            ${todo.title}
+          </li>`)}
+      </ul>
+    </div>`
+
+  function onSubmit (e) {
+    . . .
+}
+```
+
+Now every time you add an item, it should have an unchecked checkbox next to it.
+If you like, you can change the reducer to set `completed` to `true` by default,
+which will make the checkboxes show up as checked. But then change it back,
+because that isn't how our app should work.
+
+You'll notice, though, that if you add a new item, it resets all the "checked"
+statuses to the default. That's because nothing's actually happening when you
+click the checkbox other than the default browser functionality provided by
+`<input type="checkbox">`. Let's create a handler for checkbox clicks
+that fires an action to update its `completed` property in the state.
+
+The state stores todos as an array. To update a specific todo, we'll need to know
+its index in that array. So we'll alter the `state.todos.map((todos) => ...)`
+signature to include the second argument that JavaScript's `.map` function
+provides: the `index` of the array it's iterating, which we can then include in
+our `send()` call.
+
+```javascript
+const view = (state, prev, send) => {
+  return html`
+    <div>
+      <form onsubmit=${onSubmit}>
+        <input type="text" placeholder="New item" id="title">
+      </form>
+      <ul>
+        ${state.todos.map((todo, index) => html`
+          <li>
+            <input type="checkbox" ${todo.completed ? 'checked' : ''} onchange=${(e) => {
+              const updates = { completed: e.target.checked }
+              send('updateTodo', { index, updates })
+            }} />
+            ${todo.title}
+          </li>`)}
+      </ul>
+    </div>`
+
+  function onSubmit (e) {
+    . . .
+}
+```
+
+Here, we're passing `index` and an object of `updates` using the
+[ES6 property shorthand](http://es6-features.org/#PropertyShorthand) (rather than
+the redundant `{ index: index, updates: updates }`).
+
+Now we have to create a reducer called `updateTodo` to update the state when
+this action is fired.
+
+```javascript
+app.model({
+  state: {
+    todos: []
+  },
+  reducers: {
+    addTodo: (data, state) => {
+      . . .
```

Comment:
`. . .` used to confuse me a lot before I knew how to JS; like: is it real syntax? A comment would perhaps be more clear